lib/JsonNode.cpp

```diff
@@ -904,7 +904,7 @@ const JsonNode & getSchemaByName(std::string name)
 	if (vstd::contains(loadedSchemas, name))
 		return loadedSchemas[name];
 
-	std::string filename = "config/schemas/" + name + ".json";

```

Comment:
Do not change this (and all "$ref" ref fields in schemas), use
`ResourceID(std::string name, EResType::Type type)` constructor in following lines instead. It will work both with and without extension.

Reply:
My point was to change `$ref` fields, so any external app(IDE) will be able to validate mod's jsons against schema. With `vcmi:schemaName` it's not possible, because `vcmi` is not defined protocol.